lighthouse-core/runner.js

```diff
@@ -36,6 +37,13 @@ class Runner {
       return Promise.reject(err);
     }
 
+    const sentryContext = Sentry.init(opts);
```

Comment:
I was talking about the first thing, not buffering earlier events or anything (maybe we could eventually do that for CLI errors or something but "doesn't ever start" is the kind of thing that people would actually report manually, so probably isn't a pressing issue)

> this is mostly the earliest point we have all the available information to initialize

Does it need more than the `environmentData` made in `bin.ts` and `flags`? It seems like the `Sentry.init({environmentData, flags: cliFlags})` call could just be made there (or `lighthouse-core/index.js` might be better), but maybe I'm missing a step that's happening in between

Reply:
talked to brendan about this. he doesn't like that we have to pass along this environment data from file to file, augmenting method signatures along the way. the environment data in question: https://github.com/GoogleChrome/lighthouse/blob/49c061c323fcf7767d7ef04f112fb60515f6f3ec/lighthouse-cli/bin.ts#L72-L78

i'm like "sure, fine". 🦊 

**proposal:**

* move the setup of `name`, `environment`, and `release` into runner.js. no reason why it shouldn't be defined then.
* kill `tags.channel` because hey we are only recording from the cli. deal with the channels problem later when we have channels.
* restore the method signatures of runLighthouse and runner.run etc

brendan read this and is like "cool". 🐼 

@patrickhulce wdyt?

Reply:
kk, couple thoughts

the only signature that is modified is runLighthouse, runner.run is untouched
doing an fs.existsSync on .git directory directly in the node module and browser bundled code for a purely cli concern feels nastier than an extra optional argument, another consequence is that the package.json will also get pulled in as a dependency for the bundle when previously it was not 

totally cool with nuking tags.channel for now, but doesn't seem like it's hurting anything if we stick with passed in environmentData

if the signature disruption is a concern, how about we move Sentry.init to CLI land then (I believe this was brendan's original suggestion?)